gpustack/server/controllers.py

```diff
@@ -800,13 +851,24 @@ async def sync_instance_files_state(
 ):
     for file in files:
         if file.worker_id == instance.worker_id:
-            await sync_main_model_file_state(session, file, instance)
+            if (
+                instance.draft_model_source
+                and file.source_index == instance.draft_model_source.model_source_index
+            ):
+                await sync_main_worker_model_file_state(
+                    session, file, instance, is_draft_model=True
+                )
+            else:
+                await sync_main_worker_model_file_state(session, file, instance)
         else:
             await sync_distributed_model_file_state(session, file, instance)
 
 
-async def sync_main_model_file_state(
-    session: AsyncSession, file: ModelFile, instance: ModelInstance
+async def sync_main_worker_model_file_state(
```

Comment:
This function is too long. Please consider grouping it according to conditions or using other methods to enhance its readability.